### DISCUSSION TITLE
How to detect if the build failed partially

### DISCUSSION THREAD

**Discussion by ofajardo** [executablebooks/meta#128] [Q&A] [answered]
I would like to know if there is a way to either make that jupyter-book build exits with a non zero code if any if the documents failed to build or alternatively any other way to detect such a situation.

Right now in my hands if a document fails, it continues building the others and exits with a 0 status. 

The reason is that I want to create an automated report that should run periodically. If one of the documents failed I want to detect that in order to abort the deployment so that the current report is nit broken and send an email to look at the issue. 

What is the best wayto achieve this?
I have noticed that in case of error logs are written. Is that for instance reliable enough? I want to abort only if a true error happened, and also not to miss any error.
Thanks

**Comment by jstac** [comment]
We'll have the same need at https://quantecon.org/, after we switch over to JB.   @mmcky and @AakashGfude will have useful thoughts on this.

**Comment by ofajardo** [comment]
The best of course would be to have a configuration switch that would implement that behavior (full abort on any error with non 0 exit status)

**Comment by chrisjsewell** [comment, marked as answer]
Just run `jupyter-book build -W -n --keep-going mybookname/ `, 
as is already suggested in the documentation 😉 
https://jupyterbook.org/start/build.html#build-your-book-s-html

**Reply by ofajardo** [reply]
Wohoo! thanks! Exactly what I was looking for!
